shell.nix

```diff
@@ -0,0 +1,34 @@
+with import <nixpkgs> {};
```

Comment:
Can you clarify what this files is for?

Generally we try to avoid checking OS/IDE specific configuration files into the VTR source tree.

Reply:
This is so that you can type `nix-shell` and enter an [environment where all dependencies are installed](https://nixos.org/nixos/nix-pills/developing-with-nix-shell.html), ready for `make`. It's not OS or IDE specific, but does use the [Nix package manager](https://nixos.org/nix/), which runs on Linux, MacOS, and maybe Windows 10 with WSL.

I think it's worth including so that other developers can get started quickly. It's also good documentation, because `nix-shell --pure` enforces that every dependency must be listed in the .nix file by hiding all system libraries and headers.

For non-developers, I could submit a PR to [nixpkgs](https://nixos.org/nixpkgs/) at some point, but it's not clear to me that there are any non-developer users yet that would benefit.

Reply:
I've added documentation to `BUILDING.md`.